src/internal/files/files_test.go

```diff
@@ -0,0 +1,81 @@
+package files
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestFiles_WriteToJsonFile_Success(t *testing.T) {
+	dir := t.TempDir()
+
+	data := map[string]interface{}{
+		"foo": "bar",
+	}
+
+	path := filepath.Join(dir, "subdir", "file.json")
+
+	err := WriteToJsonFile(path, data)
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	raw, err := os.ReadFile(path)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var read map[string]interface{}
+	err = json.Unmarshal(raw, &read)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(data, read) {
+		t.Errorf("Expected %#v, got %#v", data, read)
+	}
+}
+
+func TestFiles_WriteToJsonFile_InvalidMarshall(t *testing.T) {
+	dir := t.TempDir()
+
+	path := filepath.Join(dir, "subdir", "file.json")
+
+	err := WriteToJsonFile(path, make(chan int))
+
+	if err == nil {
+		t.Error("Expected marshal error, got <nil>")
+	} else if !strings.Contains(err.Error(), "json: unsupported type: chan int") {
+		t.Errorf("Expected marshal error, got %s", err.Error())
+	}
+}
```

Comment:
**(nitpick)** I'd rather suggest to use cmp, or testify for such comparisons since we have them in go.mod.